server/channels/app/webhook_test.go

```diff
@@ -959,7 +979,8 @@ func TestDoOutgoingWebhookRequest(t *testing.T) {
 
 	t.Run("with a large, invalid response", func(t *testing.T) {
 		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-			io.Copy(w, InfiniteReader{})
+			_, err := io.Copy(w, InfiniteReader{})
+			require.NoError(t, err)
```

Comment:
The test fails with
```
=== FAIL: channels/app TestDoOutgoingWebhookRequest/with_a_large,_invalid_response (re-run 3) (0.00s)
    webhook_test.go:983: 
        	Error Trace:	/mattermost/server/channels/app/webhook_test.go:983
        	            				/usr/local/go/src/net/http/server.go:2171
        	            				/usr/local/go/src/net/http/server.go:3142
        	            				/usr/local/go/src/net/http/server.go:2044
        	            				/usr/local/go/src/runtime/asm_amd64.s:1695
        	Error:      	Received unexpected error:
        	            	write tcp 127.0.0.1:46157->127.0.0.1:36346: write: connection reset by peer
        	Test:       	TestDoOutgoingWebhookRequest/with_a_large,_invalid_response
```
I wonder if an error is actually expected in this case, given that the server terminates the connection.

Reply:
Shall I use ``require.Contains(t, err.Error(), "connection reset by peer")`` to ensure that the error message matches the expected behavior?

But this may not be the ideal solution. And I'm not sure whether this will work properly or not. Let me know, if you have any suggestions.

Reply:
Using `require.Contains(t, err.Error(), "connection reset by peer")` is a good idea :+1: